lucetc/src/function.rs

```diff
@@ -6,27 +6,31 @@ use cranelift_codegen::cursor::FuncCursor;
 use cranelift_codegen::entity::EntityRef;
 use cranelift_codegen::ir::{self, InstBuilder};
 use cranelift_codegen::isa::TargetFrontendConfig;
+use cranelift_frontend::FunctionBuilder;
 use cranelift_wasm::{
     FuncEnvironment, FuncIndex, GlobalIndex, GlobalVariable, MemoryIndex, SignatureIndex,
     TableIndex, WasmError, WasmResult,
 };
+use lucet_module::InstanceRuntimeData;
+use memoffset::offset_of;
 use std::collections::HashMap;
-
-// VMContext points directly to the heap (offset 0).
-// Directly before the heao is a pointer to the globals (offset -NATIVE_POINTER_SIZE).
-const GLOBAL_BASE_OFFSET: i32 = -1 * NATIVE_POINTER_SIZE as i32;
+use wasmparser::Operator;
 
 pub struct FuncInfo<'a> {
     module_decls: &'a ModuleDecls<'a>,
+    count_instructions: bool,
+    op_offsets: Vec<u32>,
     vmctx_value: Option<ir::GlobalValue>,
     global_base_value: Option<ir::GlobalValue>,
     runtime_funcs: HashMap<RuntimeFunc, ir::FuncRef>,
 }
 
 impl<'a> FuncInfo<'a> {
-    pub fn new(module_decls: &'a ModuleDecls<'a>) -> Self {
+    pub fn new(module_decls: &'a ModuleDecls<'a>, count_instructions: bool) -> Self {
         Self {
             module_decls,
+            count_instructions,
+            op_offsets: vec![0],
```

Comment:
one better, how's `scope_costs`? it was initially `op_offsets` because it started counting offsets of wasm ops for any given scope, but now it's really a stack of counters for cost when we enter a new scope

I actually think because of that cost flushing behavior in the other comment this can just be a single integer, I'm gonna think through that..